src/azul/service/elasticsearch_service.py

```diff
@@ -374,15 +374,17 @@ def _apply_paging(self,
 
         field_type = self.field_type(catalog, tuple(pagination['sort'].split('.')))
         _mode = field_type.es_sort_mode
+        _unmapped_type = getattr(field_type, 'unmapped_type', None)
 
-        def sort_values(sort_field, sort_order, sort_mode):
+        def sort_values(sort_field, sort_order, sort_mode, unmapped_type):
             assert sort_order in ('asc', 'desc'), sort_order
             return (
                 {
                     sort_field: {
                         'order': sort_order,
                         'mode': sort_mode,
-                        'missing': '_last' if sort_order == 'asc' else '_first'
+                        'missing': '_last' if sort_order == 'asc' else '_first',
+                        **({'unmapped_type': unmapped_type} if _unmapped_type is not None else {})
```

Comment:
1) Why is this referencing both the argument and the local variable?
2) Why can't we just use the local variable, since it's in scope?

Reply:
@hannes-ucsc 

1. typo from me, meant to inspect the parameter that was passed into the `sort_values` method.
2. In control flow block below (ln 398) that uses the method there are modifications that are done to `_order` parameter. However there are other parameters that are not modified within the control flow, it felt strange introducing another parameter that is not effected by the control flow but is not passed as a parameter.

Reply:
If a parameter is passed the same value in all call sites and that value is available through other means (in this case an in-scope variable) there is no need to pass the parameter.